### PR TITLE
Only show inline answers to instructors of course

### PR DESCRIPTION
**Pull request by dmcavoy**
Do not show InlineAnswers (teacher only content usually in a blue box) to participants in a course. This fixes the issue where teachers in PL courses using the normal curriculum hierarchy could see the answers for match and multi questions. Replaced spots where we were using another method to check this same thing (`can_view_teacher_markdown?`). 

One note: Currently universal instructors who are not also verified teacher can not see locked lessons. This is fine as they are not currently used in self paced PL and it will get updated in future PRs. I have added some comments in this PR to note where I should come to but I didn't want this PR to balloon up.

### Teacher Markdown & Example Solutions in Teacher Only Tab

Should have access:
Screen Shot 2021-11-30 at 9 47 32 AM

Should not have access:
Screen Shot 2021-11-30 at 9 47 45 AM

#### Free Response

Should have access:
Screen Shot 2021-11-08 at 7 06 51 AM

Should not have access:
Screen Shot 2021-11-08 at 7 07 42 AM

#### Multi

Should have access:

Screen Shot 2021-11-08 at 7 07 18 AM

Should not have access:
Screen Shot 2021-11-08 at 7 07 32 AM

#### Match

Should have access:

Screen Shot 2021-11-08 at 7 11 54 AM

Should not have access:
Screen Shot 2021-11-08 at 7 07 25 AM

### Verified Resources in Lesson Plan
Still works for existing courses (self paced pl does not have lesson plans)

Screen Shot 2021-11-08 at 7 07 50 AM

## Links

- [Eng Plan](https://docs.google.com/document/d/1V61xONU0-mleMEEdHWNxhvZtYIPRCB31mT_hbUXusCQ/edit#heading=h.fk6t1p96qqdl)
- [Product Spec](https://docs.google.com/document/d/1qZpc90daxQiiqpcQUiUiGV2tQoaYEQUzOA72UZzBlVE/edit#heading=h.qv29hwce61yz)
- [Jira](https://codedotorg.atlassian.net/browse/PLAT-1343)

## Testing story

- Manually tests Multi, Match, Free Response, Lesson Plan for existing courses and self paced pl to make sure behaves correctly ([testing doc](https://docs.google.com/spreadsheets/d/1DPfAj3Md0PprewaHATUXoXCC-FXJ0dLiNMjkotHVGo8/edit#gid=0)
- Added some new unit tests for logic for InlineAnswers

### CODE REVIEW

File: dashboard/app/models/script_level.rb [hunk 1]
```diff
@@ -743,7 +743,7 @@ def add_variant(new_level)
   def get_example_solutions(level, current_user, section_id=nil)
     level_example_links = []
 
-    return [] if !current_user&.teacher? || CDO.properties_encryption_key.blank?
+    return [] if !Policies::InlineAnswer.visible_for_script_level?(current_user, self) || CDO.properties_encryption_key.blank?
```
Comment (dmcavoy) [inline dashboard/app/models/script_level.rb:745]: Add tests for this change

File: dashboard/lib/policies/inline_answer.rb [hunk 2]
```diff
@@ -27,4 +18,22 @@ def self.visible?(user, script_level)
 
     false
   end
+
+  # Returns a boolean indicating whether or not the given user should be
+  # allowed to view answers for the given script_level
```
Comment (bethanyaconnor) [inline dashboard/lib/policies/inline_answer.rb]: nit: script

File: dashboard/test/controllers/script_levels_controller_test.rb [hunk 3]
```diff
@@ -1404,22 +1404,6 @@ def script_level_params(script_level)
 
   STUB_ENCRYPTION_KEY = SecureRandom.base64(Encryption::KEY_LENGTH / 8)
 
-  test "logged out can not view teacher markdown" do
```
Comment (bethanyaconnor) [inline dashboard/test/controllers/script_levels_controller_test.rb:1407]: do we have another test covering this (I looked briefly and couldn't find one but definitely might've missed it!)

**Review by bethanyaconnor — APPROVED** [review]
Looks good! Meta question: is "inline answers" the new term for "teacher markdown"? If not, what's the difference?

**Comment by dmcavoy** [comment]
> Looks good! Meta question: is "inline answers" the new term for "teacher markdown"? If not, what's the difference?

Inline answers was an existing name for what I usually think of as "Teacher Only" content. Before this change it controlled some but not all the Teacher Only Blue Boxes on Levels. With this change it is controlled all the Teacher Only Blue Boxes, example solutions as well as verified resources on the lesson plan. I don't love the name inline answers as I don't think it really tells you what it is talking about at this point. So I'd be open to discussing other options.

File: dashboard/test/controllers/script_levels_controller_test.rb (hunk 3 above)
Comment (dmcavoy) [reply dashboard/test/controllers/script_levels_controller_test.rb:1407]: Good call! I added tests for signed out user

**Pull request merged**